Add OpenSearch sink doc (#2311)
risingwavelabs/risingwave-docs: docs/guides/sink-to-opensearch.md

|RisingWave Data Type| OpenSearch Field Type|
|--------|--------|
|boolean |boolean|
|smallint |long|
|integer |long|
|bigint |long|
|numeric |text|
|real |float|
|double precision |float|
|character varying |text|
|bytea |text|
|date |date|
|time without time zone |text|
|timestamp without time zone | text|
|timestamp with time zone |text|
|interval |text|
|struct |object|
|array |array|
|JSONB|object (RisingWave's OpenSearch sink will send JSONB as a JSON string, and OpenSearch will convert it into an object)|

:::note
OpenSearch doesn't require users to explicitly `CREATE TABLE`. Instead, it infers the schema on-the-fly based on the first record ingested. For example, if a record contains a jsonb '{v1: 100}', v1 will be inferred as a long type. However, if the next record is '{v1: "abc"}', the ingestion will fail because "abc" is inferred as a string and the two types are incompatible.

> This behavior should be noted, or your data may be less than it should be. In terms of monitoring, you can check out Grafana, where there is a panel for all sink write errors.

Reviewer: ```suggestion
This behavior may lead to missing records. For monitoring, see Grafana, where there is a panel for all sink write errors.
```
